lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,369 @@
+    short_description: Execute modules via VMware Tools.
+    description:
+      - Execute modules via VMware Tools.
+      - "Note: Windows VMs will need to have C(ansible_shell_type: powershell) set."
+    version_added: "2.8"
+    requirements:
+      - pyvmomi (python library)
+    options:
+      connection_address:
+        description:
+          - Address for the connection (vCenter or ESXi Host).
+        vars:
+          - name: ansible_vmware_tools_connection_address
+        required: True
+      connection_username:
+        description:
+          - Username for the connection.
+          - "Requires the following permissions on the VM:
+               - VirtualMachine.GuestOperations.Execute
+               - VirtualMachine.GuestOperations.Modify
+               - VirtualMachine.GuestOperations.Query"
+        vars:
+          - name: ansible_vmware_tools_connection_username
+        required: True
+      connection_password:
+        description:
+          - Password for the connection.
+        vars:
+          - name: ansible_vmware_tools_connection_password
+        required: True
+      connection_verify_ssl:
+        description:
+          - Verify SSL for the connection.
+          - "Note: This will verify SSL for both C(connection_address) and the ESXi host running the VM."
+        vars:
+          - name: ansible_vmware_tools_connection_verify_ssl
+        default: True
+        type: bool
+      connection_ignore_ssl_warnings:
+        description:
+          - Ignore SSL warnings for the connection.
+        vars:
+          - name: ansible_vmware_tools_connection_ignore_ssl_warnings
+        default: False
+        type: bool
+      vm_path:
+        description:
+          - VM path relative to the connection.
+          - "vCenter Example: C(Datacenter/vm/Discovered virtual machine/testVM) (Needs to include C(vm) between the Datacenter and the rest of the VM path.)"
+          - "ESXi Host Example: C(ha-datacenter/vm/testVM) (Needs to include C(vm) between the Datacenter (C(ha-datacenter)) and the rest of the VM path.)"
+        vars:
+          - name: ansible_vmware_tools_vm_path
+        required: True
+      vm_username:
+        description:
+          - VM username.
+        vars:
+          - name: ansible_vmware_tools_vm_username
+        required: True
+      vm_password:
+        description:
+          - VM password.
+        vars:
+          - name: ansible_vmware_tools_vm_password
+        required: True
+      exec_command_sleep_interval:
+        description:
+          - exec command sleep interval in seconds.
+        vars:
+          - name: ansible_vmware_tools_exec_command_sleep_interval
+        default: 5
+        type: integer
+      file_chunk_size:
+        description:
+          - File chunk size.
+          - "(Applicable when writing a file to disk, example: using the C(fetch) module.)"
+        vars:
+          - name: ansible_vmware_tools_file_chunk_size
```

Comment:
Do any other connection plugins have a similar configuration?

Reply:
I don't know how many are configurable (none maybe?)... some reference `BUFSIZE` in `ansible.plugins.connection`, some are hardcoded, and others don't need it at all.